File: electric.py
import tkinter as tk
from tkinter import messagebox
import random

# Initialize main window
root = tk.Tk()
root.title("Electrical Engineering Interactive Game")
root.geometry("600x400")
root.configure(bg="#f0f0f5")

# Global score counter
score = 0

# Sample Questions and Answers
questions = [
    {
        "question": "What is the total resistance (R_total) for R1 = 5 Ω and R2 = 3 Ω in series?",
        "answer": 8,
        "hint": "In series circuits, resistances add up."
    },
    {
        "question": "What is the secondary voltage (V_s) for a primary voltage of 131 V and turns ratio of 1.57?",
        "answer": 206.04,
        "hint": "Use V_secondary = V_primary * turns ratio."
    }
]

# Function to check answer
def check_answer(question_idx):
    global score
    user_answer = entry.get()
    try:
        if float(user_answer) == questions[question_idx]["answer"]:
            score += 1
            messagebox.showinfo("Correct!", "Well done! Correct answer.")
        else:
            messagebox.showinfo("Incorrect", f"Oops! The correct answer was {questions[question_idx]['answer']}.")
        next_question()
    except ValueError:
        messagebox.showwarning("Invalid Input", "Please enter a numeric value.")

# Function to display next question
def next_question():
    global question_idx
    question_idx += 1
    if question_idx < len(questions):
        question_label.config(text=questions[question_idx]["question"])
        entry.delete(0, tk.END)
    else:
        messagebox.showinfo("Game Over", f"Final Score: {score}/{len(questions)}")
        root.destroy()

# Function to show hint
def show_hint():
    messagebox.showinfo("Hint", questions[question_idx]["hint"])

# Calculator function
def calculator():
    calc_window = tk.Toplevel(root)
    calc_window.title("Calculator")
    calc_entry = tk.Entry(calc_window, width=20, font=("Arial", 14))
    calc_entry.pack(pady=10)

    def calculate():
        try:
            result = eval(calc_entry.get())
            messagebox.showinfo("Result", f"Result: {result}")
        except:
            messagebox.showerror("Error", "Invalid expression")

    calc_button = tk.Button(calc_window, text="Calculate", command=calculate)
    calc_button.pack(pady=10)

# Main UI elements
question_idx = 0
question_label = tk.Label(root, text=questions[question_idx]["question"], font=("Arial", 14), bg="#f0f0f5", wraplength=500)
question_label.pack(pady=20)

entry = tk.Entry(root, font=("Arial", 14), width=10)
entry.pack(pady=10)

button_frame = tk.Frame(root, bg="#f0f0f5")
button_frame.pack(pady=20)

check_button = tk.Button(button_frame, text="Submit Answer", command=lambda: check_answer(question_idx))
check_button.grid(row=0, column=0, padx=5)

hint_button = tk.Button(button_frame, text="Hint", command=show_hint)
hint_button.grid(row=0, column=1, padx=5)

calc_button = tk.Button(button_frame, text="Calculator", command=calculator)
calc_button.grid(row=0, column=2, padx=5)

# Start with the first question
question_idx = 0
next_question()

# Run the main loop
root.mainloop()

File: electric.js
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.ArrayList;
import java.util.List;

public class ElectricalEngineeringGame {
    private static int score = 0;
    private static int questionIdx = 0;
    private static List<Question> questions = new ArrayList<>();
    private static JLabel questionLabel;
    private static JTextField entry;

    public static void main(String[] args) {
        // Initialize main window
        JFrame frame = new JFrame("Electrical Engineering Interactive Game");
        frame.setSize(600, 400);
        frame.getContentPane().setBackground(new Color(240, 240, 245));
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setLayout(new FlowLayout());

        // Sample Questions and Answers
        questions.add(new Question("What is the total resistance (R_total) for R1 = 5 Ω and R2 = 3 Ω in series?", 8, "In series circuits, resistances add up."));
        questions.add(new Question("What is the secondary voltage (V_s) for a primary voltage of 131 V and turns ratio of 1.57?", 206.04, "Use V_secondary = V_primary * turns ratio."));

        // Main UI elements
        questionLabel = new JLabel(questions.get(questionIdx).getQuestion());
        questionLabel.setFont(new Font("Arial", Font.PLAIN, 14));
        questionLabel.setPreferredSize(new Dimension(500, 50));
        frame.add(questionLabel);

        entry = new JTextField(10);
        entry.setFont(new Font("Arial", Font.PLAIN, 14));
        frame.add(entry);

        JPanel buttonPanel = new JPanel();
        frame.add(buttonPanel);

        JButton checkButton = new JButton("Submit Answer");
        checkButton.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                checkAnswer();
            }
        });
        buttonPanel.add(checkButton);

        JButton hintButton = new JButton("Hint");
        hintButton.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                showHint();
            }
        });
        buttonPanel.add(hintButton);

        JButton calcButton = new JButton("Calculator");
        calcButton.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                calculator();
            }
        });
        buttonPanel.add(calcButton);

        // Start with the first question
        nextQuestion();

        // Run the main loop
        frame.setVisible(true);
    }

    private static void checkAnswer() {
        String userAnswer = entry.getText();
        try {
            if (Double.parseDouble(userAnswer) == questions.get(questionIdx).getAnswer()) {
                score++;
                JOptionPane.showMessageDialog(null, "Well done! Correct answer.", "Correct!", JOptionPane.INFORMATION_MESSAGE);
            } else {
                JOptionPane.showMessageDialog(null, "Oops! The correct answer was " + questions.get(questionIdx).getAnswer() + ".", "Incorrect", JOptionPane.INFORMATION_MESSAGE);
            }
            nextQuestion();
        } catch (NumberFormatException e) {
            JOptionPane.showMessageDialog(null, "Please enter a numeric value.", "Invalid Input", JOptionPane.WARNING_MESSAGE);
        }
    }

    private static void nextQuestion() {
        questionIdx++;
        if (questionIdx < questions.size()) {
            questionLabel.setText(questions.get(questionIdx).getQuestion());
            entry.setText("");
        } else {
            JOptionPane.showMessageDialog(null, "Final Score: " + score + "/" + questions.size(), "Game Over", JOptionPane.INFORMATION_MESSAGE);
            System.exit(0);
        }
    }

    private static void showHint() {
        JOptionPane.showMessageDialog(null, questions.get(questionIdx).getHint(), "Hint", JOptionPane.INFORMATION_MESSAGE);
    }

    private static void calculator() {
        JFrame calcFrame = new JFrame("Calculator");
        calcFrame.setSize(300, 200);
        calcFrame.setLayout(new FlowLayout());

        JTextField calcEntry = new JTextField(20);
        calcFrame.add(calcEntry);

        JButton calcButton = new JButton("Calculate");
        calcButton.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                try {
                    double result = eval(calcEntry.getText());
                    JOptionPane.showMessageDialog(null, "Result: " + result, "Result", JOptionPane.INFORMATION_MESSAGE);
                } catch (Exception ex) {
                    JOptionPane.showMessageDialog(null, "Invalid expression", "Error", JOptionPane.ERROR_MESSAGE);
                }
            }
        });
        calcFrame.add(calcButton);

        calcFrame.setVisible(true);
    }

    private static double eval(String expression) {
        // Simple evaluation logic (for demonstration purposes)
        // In a real application, consider using a library for expression evaluation
        return new ScriptEngineManager().getEngineByName("JavaScript").eval(expression);
    }

    static class Question {
        private String question;
        private double answer;
        private String hint;

        public Question(String question, double answer, String hint) {
            this.question = question;
            this.answer = answer;
            this.hint = hint;
        }

        public String getQuestion() {
            return question;
        }

        public double getAnswer() {
            return answer;
        }

        public String getHint() {
            return hint;
        }
    }
}

